src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs

```diff
@@ -380,17 +396,7 @@ protected override void InitializeQosEvent()
         {
             base.InitializeQosEvent();
 
-            IAzureContext context;
-            _qosEvent.Uid = "defaultid";
-            if (RequireDefaultContext() && TryGetDefaultContext(out context))
```

Comment:
Need more time to check if it's safe to remove `RequireDefaultContext`. Will catch up next week

Reply:
@YanaXu can you test running some cmdlets that doesn't require a context after clearing the context? They should still work without a context.
Examples are the *DataCollection and ContextAutoSave cmdlets.